xlators/cluster/afr/src/afr-open.c

```diff
@@ -311,9 +325,47 @@ afr_fix_open(fd_t *fd, xlator_t *this)
     ret = loc_path(&local->loc, NULL);
     if (ret < 0)
         goto out;
+    local->cont.lk.getlk_rsp = GF_CALLOC(sizeof(*local->cont.lk.getlk_rsp),
+                                         priv->child_count, gf_afr_mt_gf_lock);
 
     local->fd = fd_ref(fd);
     local->fd_ctx = fd_ctx;
+    local->strict_locks = _gf_false;
+    flock.l_type = F_WRLCK;
+    afr_set_lk_owner(frame, this, frame->root);
+    flock.l_owner.len = frame->root->lk_owner.len;
+    if (flock.l_owner.len > 0) {
+        memcpy(flock.l_owner.data, frame->root->lk_owner.data,
+               flock.l_owner.len);
+    }
+
+    AFR_ONLIST(priv->child_up, frame, afr_getlk_cbk, lk, fd, F_GETLK, &flock,
+               xdata);
+
+    success_replies = alloca0(priv->child_count);
+    afr_fill_success_replies(local, priv, success_replies);
+    if (AFR_COUNT(success_replies, priv->child_count) == 0) {
+        ret = -1;
+        goto out;
+    }
+
+    for (i = 0; i < priv->child_count; i++) {
+        if (!local->replies[i].valid || local->replies[i].op_ret != 0)
+            continue;
+        if (local->cont.lk.getlk_rsp[i].l_type == F_UNLCK) {
+            continue;
+        } else if (local->cont.lk.getlk_rsp[i].l_type == F_WRLCK) {
+            locks_held = _gf_true;
```

Comment:
Can we still allow open if the lock present on the up bricks are still from the same fd and client?

Reply:
Added check for lk_owner.

Reply:
```afr-open.c: In function ‘afr_fix_open’:
afr-open.c:360:50: warning: passing argument 2 of ‘is_same_lkowner’ from incompatible pointer type [-Wincompatible-pointer-types]
  360 |                                     flock.l_owner.data)) {
      |                                     ~~~~~~~~~~~~~^~~~~
      |                                                  |
      |                                                  char *
In file included from ../../../../libglusterfs/src/glusterfs/glusterfs.h:38,
                 from afr-open.c:16:
../../../../libglusterfs/src/glusterfs/lkowner.h:63:49: note: expected ‘gf_lkowner_t *’ but argument is of type ‘char *’
   63 | is_same_lkowner(gf_lkowner_t *l1, gf_lkowner_t *l2)
      |                                   ~~~~~~~~~~~~~~^~```

Reply:
Since we should not re-open the fd even if it is from the same owner, removed this check.